upup/pkg/fi/cloudup/openstacktasks/subnet.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openstacktasks
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/subnets"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/openstack"
+)
+
+//go:generate fitask -type=Subnet
+type Subnet struct {
+	ID        *string
+	Name      *string
+	Network   *Network
+	CIDR      *string
+	Lifecycle *fi.Lifecycle
+}
+
+var _ fi.CompareWithID = &Subnet{}
+
+func (s *Subnet) CompareWithID() *string {
+	return s.ID
+}
+
+func (s *Subnet) Find(context *fi.Context) (*Subnet, error) {
+	cloud := context.Cloud.(openstack.OpenstackCloud)
+	opt := subnets.ListOpts{
+		ID:         fi.StringValue(s.ID),
+		Name:       fi.StringValue(s.Name),
+		NetworkID:  fi.StringValue(s.Network.ID),
+		CIDR:       fi.StringValue(s.CIDR),
+		EnableDHCP: fi.Bool(true),
```

Comment:
I'm not 100% familiar with the Neutron API, but this seems like it might be selecting too narrowly.  Typically you want to find the network by one piece of information (the most specific you can), and then you return the values that you found.  And - for example - if we want DHCP to be enabled, we would set that as a field on `Subnet`, we would return the actual value we found, and then `RenderOpenstack` would be responsible for setting the DHCP value if it was changed.

But ... not sure about this, so going to LGTM anyway!